rpc/src/services/protocol/proto_005_2/contract_service.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use std::string::ToString;
+
+use failure::bail;
+
+use storage::skip_list::Bucket;
+use storage::context::{TezedgeContext, ContextIndex, ContextApi};
+use tezos_messages::base::signature_public_key::SignaturePublicKey;
+use tezos_messages::p2p::binary_message::BinaryMessage;
+
+use crate::helpers::ContextProtocolParam;
+use crate::services::protocol::proto_005_2::helpers::{construct_indexed_contract_key};
+
+pub(crate) fn get_contract_counter(context_proto_params: ContextProtocolParam, pkh: &str, context: TezedgeContext) -> Result<Option<String>, failure::Error> {
+
+    // level of the block
+    let level = context_proto_params.level;
+
+    let context_index = ContextIndex::new(Some(level), None);
+    
+    let indexed_contract_key = construct_indexed_contract_key(pkh)?;
+
+    // ["data","contracts","index","91","6e","d7","72","4e","49","0000535110affdb82923710d1ec205f26ba8820a2259","counter"]
+    let contract_counter;
```

Comment:
@adonagy 
this can be written in one-line:
```
let contract_counter = if let Some(Bucket::Exists(data)) = context.get_key(&context_index, &contract_counter_key)? {
    contract_counter = Some(tezos_messages::protocol::proto_005_2::contract::Counter::from_bytes(data)?)
} else {
    contract_counter = None
};
```